lib/core/viewmodels/dashboard_viewmodel.dart

```diff
@@ -101,11 +118,66 @@ class DashboardViewModel extends FutureViewModel<Map<PreferencesFlag, int>> {
       orderedCards.forEach((key, value) {
         if (value >= 0) {
           _cardsToDisplay.insert(value, key);
+          if (key == PreferencesFlag.scheduleCard) {
+            futureToRunSchedule();
+          }
         }
       });
     }
   }
 
+  /// Return the list of all the courses activities arranged by date.
+  Map<DateTime, List<CourseActivity>> get coursesActivities {
```

Comment:
Do you really need this function? You could pass through _courseRepository.coursesActivities and only recover the activities for today, no?

Reply:
Yup it was an overkill amount of logic, I changed that part to remove this function and only get today's activities instead as suggested.